tests/BuiltInWebServerAbstract.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+namespace atk4\ui\tests;
+
+use GuzzleHttp\Client;
+use PHPUnit\Framework\TestCase;
+use Psr\Http\Message\ResponseInterface;
+use Symfony\Component\Process\Process;
+
+abstract class BuiltInWebServerAbstract extends TestCase
+{
+    protected static $process;
+
+    protected static $host = '127.0.0.1';
+    protected static $port = 9687;
+
+    /** @var bool set the app->call_exit in demo */
+    protected static $app_def_call_exit = true;
+
+    /** @var bool set the app->caught_exception in demo */
+    protected static $app_def_caught_exception = true;
+
+    protected static $webserver_root = 'demos/';
+
+    public static function setUpBeforeClass()
+    {
+        if (!file_exists(getcwd().'/coverage/')) {
```

Comment:
to get coverage on tests run with built in server i use the same idea that we already use with selenium, include file coverage.php and collect .cov files